### ISSUE TITLE
Tutorial (training) problems

### ISSUE THREAD

**Issue by amadanmath** [PAIR-code/deeplearnjs#22]
First off, while I understand the basic principles of NN, I must admit to being rather new to working with them, so maybe my problems are trivial.

I tried following the [tutorial](https://pair-code.github.io/deeplearnjs/docs/tutorials/intro.html), and copied the code [to jsbin](https://jsbin.com/pexojojesu/edit?html,js,output). The code is changed minimally in the attempt to make it work: assumed `labels` should be an array of shape `[1]` NDArrays as declared in `labelShape` and not array of shape `[3]` NDArrays as initialised in `const labels`, and that the second `const inputTensor` should really be `const labelTensor`. (Also, trivially, I removed type annotations and added the namespace to make it run directly in JavaScript.)

However, I can't get anything close to the expected label from this code.

I tried increasing the number of examples to 1000 (random autogenerated, with calculated labels). I tried increasing the batch size. I tried training in a loop. It all ended badly (i.e. with huge cost, and prediction very far from the expected value).

I tried to look up the documentation on `Session.prototype.train`; I couldn't work out which data is being processed, how many iterations are being done, or whether repeating it would add more training data or start from scratch (and if training is meant to be repeated, does the data need to be reshuffled)...

All in all, would it be possible to add an executable version of the training tutorial so we can see it in action (and how to write it correctly)? I can see the library itself works in Demos, but those are a bit too complex for basic study.

Thank you

**Comment by nsthorat** [comment]
Thanks for the comment, sorry about that! Will fix this tonight (currently traveling).

**Comment by nsthorat** [comment]
Okay, so I had a bug in the inputs / label definitions (we have another bug right now, so I'm waiting for that to get fixed before I commit the intro.ts tutorial code to head). This is what the inputs / labels should be:

`ts
  const inputs: Array1D[] = [
    Array1D.new([1.0, 2.0, 3.0]),
    Array1D.new([10.0, 20.0, 30.0]),
    Array1D.new([100.0, 200.0, 300.0])
  ];

  const labels: Array1D[] = [
    Array1D.new([2.0]),
    Array1D.new([20.0]),
    Array1D.new([200.0])
  ];
`

There was also a bug where I redefined "inputTensor", when the second one should be "labelTensor". I thought this would be trivial and I wouldn't have to actually run it, guess I was totally wrong. I will update this ASAP for you. So sorry about this!

**Comment by nsthorat** [comment]
For reference, I'm waiting for this fix: https://github.com/PAIR-code/deeplearnjs/pull/30
